backend/app/email_verification.py

```
@@ -0,0 +1,117 @@
+
+def hash_verification_code(code: str) -> str:
+    raw = (code or "").strip()
+    return hmac.new(_signing_key(), raw.encode("utf-8"), hashlib.sha256).hexdigest()
+
+
+def verification_codes_match(plain: str, stored_hash: str | None) -> bool:
+    if not stored_hash:
+        return False
+    try:
+        return hmac.compare_digest(hash_verification_code(plain), stored_hash)
+    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)